src/core/shapes/Union.hpp

```diff
@@ -66,6 +67,14 @@ class Union : public Shape {
                         dist_compare);
   }
 
+  bool is_inside(Utils::Vector3d const &pos) const override {
+    return boost::accumulate(
```

Comment:
I think you should use `any_of`, which will short-cut and makes the intent clearer.

Reply:
you probably mean ```all_of```